src/adaXT/decision_tree/DecisionTree.pyx

```diff
@@ -44,6 +46,41 @@ class DecisionTree:
         self.classes = None
         self.splitter = splitter
 
+    def __error_check_max_features(self, max_features):
+        if max_features is None:
+            return max_features
+        elif isinstance(max_features, int):
+            if max_features < 1:
+                raise ValueError("max_features can not be less than 1")
+            else:
+                return max_features
+        elif isinstance(max_features, float):
+            return max_features
```

Comment:
Should we still not make sure it is non negative even when it is a float? Perhaps even check
that the value of the float is between 0 and 1.

Reply:
Done.